src/Nethermind/Nethermind.Core.Test/Encoding/BlockInfoDecoderTests.cs

```diff
@@ -60,5 +65,40 @@ private static void Roundtrip(bool valueDecode)
             Assert.AreEqual(TestItem.KeccakA, decoded.BlockHash, "block hash");
             Assert.AreEqual(UInt256.One, decoded.TotalDifficulty, "difficulty");
         }
+
+        private static void RoundtripBackwardsCompatible(bool valueDecode, bool chainWithFinalization, bool isFinalized)
+        {
+            BlockInfo blockInfo = new(TestItem.KeccakA, 1);
+            blockInfo.WasProcessed = true;
+            blockInfo.IsFinalized = isFinalized;
+
+            Rlp rlp = BlockInfoEncodeDeprecated(blockInfo, chainWithFinalization);
+            BlockInfo decoded = valueDecode ? Rlp.Decode<BlockInfo>(rlp.Bytes.AsSpan()) : Rlp.Decode<BlockInfo>(rlp);
+
+            Assert.True(decoded.WasProcessed, "0 processed");
+            Assert.AreEqual(chainWithFinalization && isFinalized, decoded.IsFinalized, "finalized");
+            Assert.AreEqual(TestItem.KeccakA, decoded.BlockHash, "block hash");
+            Assert.AreEqual(UInt256.One, decoded.TotalDifficulty, "difficulty");
+        }
+
+        public static Rlp BlockInfoEncodeDeprecated(BlockInfo? item, bool chainWithFinalization)
+        {
+            if (item == null)
+            {
+                return Rlp.OfEmptySequence;
+            }
+
+            Rlp[] elements = new Rlp[chainWithFinalization ? 4 : 3];
+            elements[0] = Rlp.Encode(item.BlockHash);
+            elements[1] = Rlp.Encode(item.WasProcessed);
+            elements[2] = Rlp.Encode(item.TotalDifficulty);
+
+            if (chainWithFinalization)
+            {
+                elements[3] = Rlp.Encode(item.IsFinalized);
```

Comment:
test encoding with multiple flags?

Reply:
This method is for encoding in the way that it was done before added the metadata field, it is only used to test that the new decoder is backwards compatible, i.e. `BlockInfo` encoded before adding metadata can be decoded by the new decoder. However I added a few Metadata flags to the roundtrip test above.